Engine/Component/ComponentParticleSystem.cpp

```diff
@@ -24,6 +25,8 @@ ComponentParticleSystem::~ComponentParticleSystem()
 
 void ComponentParticleSystem::Init() 
 {
+	glGenBuffers(1, &ssbo);
+	shader_program = App->program->GetShaderProgramId("Particles");
```

Comment:
if you are going for the approach of creating one shader for PS, then you should send the uniforms on the componentParticleSysyem, not on billboard

Reply:
but this way I will be duplicating code.